quipucords/fingerprinter/task.py

```diff
@@ -338,6 +338,15 @@ def _process_details_report(self, manager_interrupt, details_report):
                     for fact in CANONICAL_FACTS:
                         if fingerprint_dict.get(fact):
                             found_canonical_facts = True
+                            not_null_facts = {
```

Comment:
I would move this definition to line 355 so that it is clear where it is being used.